core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
encode_track_it_block_di() needs to know whether this is OP_it, so it seems like it has to call the decoder from this encode routine if the instr is undecoded.

For instructions that the manual calls "unpredictable": see #1685.